lib/sinon/assert.js

```diff
@@ -333,4 +333,20 @@ function createAssertObject(opts) {
 }
 
 module.exports = createAssertObject();
-module.exports.createAssertObject = createAssertObject;
+Object.defineProperty(module.exports, "createAssertObject", {
+    get() {
+        return createAssertObject;
+    },
+});
+
+// allow for easy mocking
+module.exports.mock = {};
+Object.defineProperty(module.exports.mock, "createAssertObject", {
```

Comment:
I'm not a fan of this approach, as it adds test code to production code.

For a testing library, adding some bloat might not be that big a deal ... but for production code, especially code that ships over the internet, this adds a lot of bloat of unused code. (No, tree shaking is not a good argument to embrace this practice).

Additionally, if the code enabling tests is not added very carefully, it becomes entangled with the production code.



Since https://github.com/sinonjs/sinon/commit/11a8e3f24ccec981a280dbe63df316d1fbde4ddd, where proxyquire was removed, this seems to be one solution ... are there others?

I wish we had at least had a discussion about proxyquire and prettier conflict, so we could have evaluated solutions before they got merged into `main`.

Reply:
👍 @mroderick . I got rid of the `mock` object. It was not necessary at all, as I got this a bit confused with ESM exports for a while, and seeing it work made me realize the whole getter thing was just a totally unneccessary